kit/Kit.cpp

```diff
@@ -3313,6 +3317,37 @@ void lokit_main(
                     return false;
                 }
 
+                if (!FileUtil::isEmptyDirectory(loJailDestImpressTemplatePath)) // make sure template directory exist
```

Comment:
Maybe you don't need this "isEmptyDirectory" condition at all if https://gerrit.libreoffice.org/c/core/+/177353 is merged ?

But if you do, maybe it should be something like:

const FileUtil::Stat templateStat(loJailDestImpressTemplatePath);
if (templateStat.good() && templateStat.isDirectory())  // make sure template directory exist

Because if the directory exists, but it is empty, then isEmptyDirectory is true and this code isn't run, but an empty target dir here is perfectly ok as long as it exists.

After that (with my core patch above applied), if I then remove the explicit copy of "copy default tempates from 'common' dir" I think it works ok without it.

The mount namespaces can get a bit confusing, but FWIW given the pid of the kit, e.g. from a log message like: kit-2421928-2421928 2024-11-26 17:08:44.948975 +0000 [ kit_spare_002 ] DBG  Mounting...

you can:

$ sudo nsenter -m -t 2421928

and then mount and ls will report what pid 2421928 sees.

So for me mount reports:

/dev/nvme0n1p3 on /home/caolan/LibreOffice/collabora-online/jails/2421849-b13c8f67/5pFZQab2t34k3MK1/lo/share/template/common/presnt ...

so I can see something got mounted on the lo/share/template/common/presnt dir and

ls /home/caolan/LibreOffice/collabora-online/jails/2421849-b13c8f67/5pFZQab2t34k3MK1/lo/share/template/common/presnt

lists

template1.otp  template2.otp  template3.otp

So that looked like it worked.

We do want this "copy" fallback though for the case that bindMounts fail, I think that should go further down after the "linkOrCopy(loTemplate, loJailDestPath" lines which is where we copy the libreoffice install when we can't mount so this could go after that.